### PR TITLE
[BACKLOG-39521] Added if statement to remove username and password be…

### PR DESCRIPTION
**Pull request by MarcDiamantHitachi**
…ing printed

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 8s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl tomcat-logs
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 46s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl tomcat-logs
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 42s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl tomcat-logs
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: tomcat-logs/src/main/java/org/pentaho/tomcat/logvalve/FilteredAccessLogValve.java [hunk 1]
```diff
@@ -35,10 +35,18 @@ public class FilteredAccessLogValve extends AccessLogValve {
...
     try ( CharArrayWriter caw = new CharArrayWriter() ) {
       // Mask the user password
-      caw.write( message.toString().replaceAll( "j_password=[^&^ ]*", "j_password=***" ) );
+      String tempString = message.toString();
+      if ( tempString.contains( "/pentaho/api/csrf" ) || tempString.contains( "/pentaho/api/repo/files/backup" ) ) {
+        tempString = tempString.replaceAll( "\\?userid[^&]+%26", "" );
+        tempString = tempString.replaceAll( "\\?userid[^&]+", "" );
+        tempString = tempString.replaceAll( "password[^&]+%26", "" );
+        tempString = tempString.replaceAll( "\\&password[^&]+", "" );
+      }
```
Comment (dcleao) [inline tomcat-logs/src/main/java/org/pentaho/tomcat/logvalve/FilteredAccessLogValve.java:44]: IMO, this implementation should not have been merged as is.

Why does the filter only apply to these two URLs?
The parameter names used by `RequestParameterAuthenticationFilter` are configurable via the properties of the `requestParameterProcessingFilter` bean. Some customers may have these configured differently.

**Pull request merged**